src/store/ChartStore.js

```diff
@@ -279,6 +280,18 @@ class ChartStore {
 
         stxx.append('createDataSet', this.updateComparisons);
     }
+    initialRouting() {
+        let timer;
+        window.addEventListener('hashchange', () => {
+            clearTimeout(timer);
+            timer = setTimeout(() => {
+                const hash = window.location.hash.replace('#', '');
+                if (hash === '') {
+                    this.mainStore.dialog.closeAll();
+                }
+            }, 300);
+        }, false);
+    }
```

Comment:
Move this to a new store named `RoutingStore`, create an instance of it in `store/index.js` and have all the DailogStore instances register themselves with the new RoutingStore.

```javascript
class RoutingStore {
   allDialogs = [];
   registerDialog(dialogStore) {  this.allDialogs.push(dialogStore); }
   constructor() {
      let timer;
      window.addEventListener( ...
     /...
    }
}
```